js/XAPI.js

```diff
@@ -65,6 +65,8 @@ class XAPI extends Backbone.Model {
       components: 'component',
       offlineStorage: 'offlineStorage'
     };
+
+    this.sendState = _.debounce(this.sendState.bind(this), 500);
```

Comment:
Sorry, I forgot to articulate a problem with this in my ramblings.

`_.debounce` holds back multiple calls to a function until a given time period has passed, then it calls the function only once, with only the last set of arguments passed into the debounce queue.

I believe `sendState` is doing two things, updating a client side state object and then sending that state object to the server.

If this is the case, the first part of `sendState`, updating the client side state object, cannot be hidden behind a debounce as the function requires every set of arguments sent to perform its client side state object updates. The latter part of the `sendState` function can be debounced, as server updates only need to be sent periodically.

Reply:
Ah very good point, I will amend the code. Thanks